coordinator/src/main/scala/com/socrata/datacoordinator/service/collocation/Cost.scala

```diff
@@ -22,11 +18,41 @@ case class Cost(moves: Int, totalSizeBytes: Long, moveSizeMaxBytes: Option[Long]
 object Cost {
   implicit val encode = AutomaticJsonEncodeBuilder[Cost]
 
-  def max(x: Cost, y: Cost): Cost = Ordering[Cost].max(x, y)
-
   val Zero: Cost = Cost(moves = 0, totalSizeBytes = 0L, moveSizeMaxBytes = Some(0L))
 
   // since we have not yet backfilled the secondary_metrics table and we don't want to
   // outright fail, return an obviously not real value for the total size in bytes.
   val Unknown: Cost = Cost(moves = 1, totalSizeBytes = -1L)
 }
+
+// Parameters must all be non-negative and
+// for Cost.Zero < Cost.Unknown we must have:
+//
+// movesWeight > totalSizeBytesWeight + moveSizeMaxBytesWeight
+// and movesWeight > 0.0
+case class WeightedCostOrdering(movesWeight: Double,
+                                totalSizeBytesWeight: Double,
+                                moveSizeMaxBytesWeight: Double) extends Ordering[Cost] {
+
+  if (movesWeight <= 0.0) throw new IllegalArgumentException("movesWeight must be positive")
+
+  private def parameterNonNegative(name: String, value: Double): Unit =
```

Comment:
This is probably overkill since you're not really stopping much code reuse.
`if (totalSizeBytesWeight < 0.0) throw new IllegalArgumentException("totalSizeBytesWeight must be non-negative")` reads much easier.